spec/autocomplete-manager-integration-spec.coffee

```diff
@@ -336,7 +335,7 @@ describe 'Autocomplete Manager', ->
               charMatch = editorView.querySelector('.autocomplete-plus li span.word .character-match')
               expect(word.textContent).toBe 'ab(c)c'
               expect(charMatch.textContent).toBe 'c'
-              expect(charMatch.parentNode).toHaveClass 'word'
+              expect(charMatch.parentNode.parentNode).toHaveClass 'word'
```

Comment:
@benogle I think the tests you have are invalid. It should be valid to have a character match for something within the snippet body. It's just not valid to match any of the HTML wrapping a snippet (i.e. `<span class="snippet-completion">` or `</span>`) or the snippet definition (i.e. `${0:` or `}`).

With my new logic, I'm able to stably determine the snippet definition text and exclude it from character matches, prior to insertion of snippet HTML. Consequently, the tests need to change. I think the intent of #301 is to ensure that:

1. Snippet HTML does not make its way into the suggestion list
1. Snippet HTML is not highlighted
1. Snippet definition text does not make its way into the suggestion list
1. Snippet definition text does not cause incorrect highlighting of character matches

This is now achieved with my latest changes, that I am about to push.